### ISSUE TITLE
`setSignatureAsync` function not working for Outlook Mac 

### ISSUE THREAD

**Issue by andreyvcita** [OfficeDev/office-js#2335]
`setSignatureAsync` worked a few days after release [1.10 New Mac UI now supports](https://github.com/OfficeDev/office-js-docs-pr/pull/3130)

* Platform: Mac 
* Host [Excel, Word, PowerPoint, etc.]: Outlook (new UI)
* Office version number: 16.56
* Operating System: MacOS 12.0.1
 
## Expected behavior
`setSignatureAsync` set signature 

## Current behavior
`setSignatureAsync` not working

**Comment by Wenjun-Gong** [comment]
Thanks for reporting this issue. Assign to @exextoc to take a look.

**Comment by exextoc** [comment]
Hey @andreyvcita ,

We are unable to reproduce this issue. Can you go to Help -> Collect Diagnostics, which will help us collect logs for your session and then share the Diagnostics ID with us so that we can look into it?

**Comment by leewisestamp** [comment]
Hi @exextoc, here is @andreyvcita's session info (we're on the same team):

![image](https://user-images.githubusercontent.com/11390801/147663416-21d213f9-80be-4408-88bc-617e8a34e12e.png)
![image](https://user-images.githubusercontent.com/11390801/147663097-31dfbfb6-ce25-4845-b42c-b610f80022eb.png)

**Comment by leewisestamp** [comment]
Any update on this, please?

**Comment by exextoc** [comment]
@andreyvcita We tried to reproduce the issue with the add-in provided by you through the email. The add-in works for my account.
Therefore, could you try the following.
1. Can you add the following test account(credentials sent via email) in your mac client and see if the add-in works there.
2. If this does not work. Could you try to update the outlook to the latest version and check.
Let us know your findings.

**Comment by andreyvcita** [comment]
signature: 6284 characters

```js
Office.context.mailbox.item.body.setSignatureAsync(signature, {coercionType: Office.CoercionType.Html},
    function (asyncResult) {
        if (asyncResult.status === Office.AsyncResultStatus.Succeeded) {
            console.log("setSignatureAsync succeeded");
        } else {
            console.log("setSignatureAsync failed with error: " + JSON.stringify(asyncResult.error));
        }
});
```
> setSignatureAsync failed with error: {"name":"Invalid Format Error","message":"The format of the specified data object is invalid.","code":2006}
user-script:18:18

I only get this error on Mac

the same code with function setAsync() works

**Comment by andreyvcita** [comment]
looks like it's not working through limit, because it works for signature 3430 characters

**Comment by leewisestamp** [comment]
@exextoc - It looks like the underlying problem is that Outlook Mac handles the HTML passed to `setSignatureAsync` differently from Outlook Web/Windows. When we tried calling setSignatureAsync with very basic HTML (e.g `<span>Hello, world!</span>`) in Outlook Mac, it worked. But, when we tried with more complex HTML, Outlook Mac behaved differently (i.e. rejected the HTML) than Outlook Web/Windows.

I think this difference in behavior should be classified as a bug: either a product bug that needs fixing, or a documentation bug that needs clarification.

**Comment by leewisestamp** [comment]
> @andreyvcita We tried to reproduce the issue with the add-in provided by you through the email. The add-in works for my account. Therefore, could you try the following.
> 
> 1. Can you add the following test account(credentials sent via email) in your mac client and see if the add-in works there.
> 2. If this does not work. Could you try to update the outlook to the latest version and check.
>    Let us know your findings.

@exextoc - It appears the underlying problem is Outlook Mac handles HTML passed to `setSignatureAsync` differently from Outlook Web/Windows. When calling `setSignatureAsync` with very basic HTML (e.g <span>Hello, world!</span>) in Outlook Mac, it worked. But, when trying with more complex HTML (specifically, HTML including carriage returns), Outlook Mac rejected the HTML as opposed to Outlook Web/Windows which accepted it.

HTML that works in Web/Windows but not in Mac:

```
<div><table><tr><td><table><tr><td><table><tr><td><div>
    
    
    
        
        
            
            
    <a><img style="display:block;height:57px" height="57" src="https://s3.amazonaws.com/apps.wisestamp.com/assets/signoff/k_regards.gif"></a>
            
            
        
    
</div>

</td></tr></table></td></tr></table></td></tr></table></div>
```

I think this difference in behavior should be classified as a bug: either a product bug that needs fixing, or a documentation bug that needs clarification.

**Comment by exextoc** [comment]
Thank you for elaborating your issue. It is currently a known issue that if you add a newline character to the html body, the SetSignature API throws a validation error. The fix shall be available in next Outlook Mac version.

**Comment by leewisestamp** [comment]
Thanks for your prompt and thorough replies! Eagerly awaiting the fix. Cheers!

**Comment by barclayadam** [comment]
I have run into a similar problem on Mac (v16.56.1211.0, I also created an incident with ID C4BJ2CBZ), but it does not appear to be because of newlines. I can set complex HTML signatures via. `setSignature`, but only if I do not set `coercionType`:

Either of the below calls work:

````js
Office.context.mailbox.item.body.setSignatureAsync(body);
Office.context.mailbox.item.body.setSignatureAsync(body, { coercionType: undefined });
````

This does not:

````js
Office.context.mailbox.item.body.setSignatureAsync(body, { coercionType: 'html' });
````

This code works absolutely fine OWA and Windows. On Mac I get the error `{"name":"Invalid Format Error","message":"The format of the specified data object is invalid.","code":2006}`

**Comment by barclayadam** [comment]
I have continued to test this and have noticed that the behaviour is different depending on whether this method is called from the task pane or event-based activation.

The task pane will fail even with the change to not pass `coercionType`, whilst event-based activation will allow the HTML to be set regardless of newlines or not

**Comment by exextoc** [comment]
@barclayadam The fix mentioned above would solve your issue too. Kindly wait for the next release of the Outlook Mac.

**Comment by exextoc** [comment]
@andreyvcita We have fixed the bug, and the fix is available in the latest build 16.57. Kindly test it and let us know your feedback, please.

**Comment by andreyvcita** [comment]
@exextoc It works! Thanks!
